main.go

```diff
@@ -0,0 +1,586 @@
+	:m_rows_examined_p99,
+	:m_rows_examined_hg,
+	:m_rows_affected_cnt,
+	:m_rows_affected_sum,
+	:m_rows_affected_min,
+	:m_rows_affected_max,
+	:m_rows_affected_p99,
+	:m_rows_affected_hg,
+	:m_rows_read_cnt,
+	:m_rows_read_sum,
+	:m_rows_read_min,
+	:m_rows_read_max,
+	:m_rows_read_p99,
+	:m_rows_read_hg,
+	:m_merge_passes_cnt,
+	:m_merge_passes_sum,
+	:m_merge_passes_min,
+	:m_merge_passes_max,
+	:m_merge_passes_p99,
+	:m_merge_passes_hg,
+	:m_innodb_io_r_ops_cnt,
+	:m_innodb_io_r_ops_sum,
+	:m_innodb_io_r_ops_min,
+	:m_innodb_io_r_ops_max,
+	:m_innodb_io_r_ops_p99,
+	:m_innodb_io_r_ops_hg,
+	:m_innodb_io_r_bytes_cnt,
+	:m_innodb_io_r_bytes_sum,
+	:m_innodb_io_r_bytes_min,
+	:m_innodb_io_r_bytes_max,
+	:m_innodb_io_r_bytes_p99,
+	:m_innodb_io_r_bytes_hg,
+	:m_innodb_io_r_wait_cnt,
+	:m_innodb_io_r_wait_sum,
+	:m_innodb_io_r_wait_min,
+	:m_innodb_io_r_wait_max,
+	:m_innodb_io_r_wait_p99,
+	:m_innodb_io_r_wait_hg,
+	:m_innodb_rec_lock_wait_cnt,
+	:m_innodb_rec_lock_wait_sum,
+	:m_innodb_rec_lock_wait_min,
+	:m_innodb_rec_lock_wait_max,
+	:m_innodb_rec_lock_wait_p99,
+	:m_innodb_rec_lock_wait_hg,
+	:m_innodb_queue_wait_cnt,
+	:m_innodb_queue_wait_sum,
+	:m_innodb_queue_wait_min,
+	:m_innodb_queue_wait_max,
+	:m_innodb_queue_wait_p99,
+	:m_innodb_queue_wait_hg,
+	:m_innodb_pages_distinct_cnt,
+	:m_innodb_pages_distinct_sum,
+	:m_innodb_pages_distinct_min,
+	:m_innodb_pages_distinct_max,
+	:m_innodb_pages_distinct_p99,
+	:m_innodb_pages_distinct_hg,
+	:m_query_length_cnt,
+	:m_query_length_sum,
+	:m_query_length_min,
+	:m_query_length_max,
+	:m_query_length_p99,
+	:m_query_length_hg,
+	:m_bytes_sent_cnt,
+	:m_bytes_sent_sum,
+	:m_bytes_sent_min,
+	:m_bytes_sent_max,
+	:m_bytes_sent_p99,
+	:m_bytes_sent_hg,
+	:m_tmp_tables_cnt,
+	:m_tmp_tables_sum,
+	:m_tmp_tables_min,
+	:m_tmp_tables_max,
+	:m_tmp_tables_p99,
+	:m_tmp_tables_hg,
+	:m_tmp_disk_tables_cnt,
+	:m_tmp_disk_tables_sum,
+	:m_tmp_disk_tables_min,
+	:m_tmp_disk_tables_max,
+	:m_tmp_disk_tables_p99,
+	:m_tmp_disk_tables_hg,
+	:m_tmp_table_sizes_cnt,
+	:m_tmp_table_sizes_sum,
+	:m_tmp_table_sizes_min,
+	:m_tmp_table_sizes_max,
+	:m_tmp_table_sizes_p99,
+	:m_tmp_table_sizes_hg,
+	:m_qc_hit_sum,
+	:m_full_scan_sum,
+	:m_full_join_sum,
+	:m_tmp_table_sum,
+	:m_tmp_table_on_disk_sum,
+	:m_filesort_sum,
+	:m_filesort_on_disk_sum,
+	:m_select_full_range_join_sum,
+	:m_select_range_sum,
+	:m_select_range_check_sum,
+	:m_sort_range_sum,
+	:m_sort_rows_sum,
+	:m_sort_scan_sum,
+	:m_no_index_used_sum,
+	:m_no_good_index_used_sum,
+	:grpstr,
+	:grpint,
+	:labint_key
+	:labint_value
+  )
+`
+
+// Save inserts parsed query into ClickHouse DB
+func (qc *QueryClass) Save(e *slowlog.Event) (rerr error) {
+	tx, err := qc.db.Beginx()
+	if err != nil {
+		return fmt.Errorf("transaction begin error: %s", err.Error())
+	}
+
+	stmt, err := tx.PrepareNamed(insertSQL)
+	if err != nil {
+		return fmt.Errorf("prepare error: %s", err.Error())
+	}
+	defer func() {
+		if e := stmt.Close(); e != nil {
```

Comment:
The statement should be closed before tthe ransaction is committed:
* begin transaction;
* prepare statement;
* execute statement X times;
* release statement;
* commit transaction.